include/picongpu/plugins/transitionRadiation/TransitionRadiation.x.cpp

```diff
@@ -105,31 +204,43 @@ namespace picongpu
                 std::vector<complex_X> tmpCTRperp;
                 std::vector<float_X> tmpNum;
                 std::vector<float_X> theTransRad;
-                MappingDesc* cellDescription = nullptr;
-                std::string notifyPeriod;
                 uint32_t timeStep;
 
                 std::string pluginName;
                 std::string speciesName;
                 std::string pluginPrefix;
-                std::string folderTransRad;
                 std::string filenamePrefix;
+                std::string fileExtension;
 
                 bool isMaster = false;
                 uint32_t currentStep = 0;
 
                 mpi::MPIReduce reduce;
 
+                MappingDesc* m_cellDescription = nullptr;
+                std::shared_ptr<Help> m_help;
+                size_t m_id;
+
+                bool textOutput;
+                float_X foilPositionYSI;
+
             public:
                 //! Constructor
-                TransitionRadiation()
-                    : pluginName("TransitionRadiation: calculate transition radiation of species")
-                    , speciesName(T_ParticlesType::FrameType::getName())
-                    , pluginPrefix(speciesName + std::string("_transRad"))
-                    , folderTransRad("transRad")
-                    , filenamePrefix(pluginPrefix)
+                TransitionRadiation(
```

Comment:
For some reason, one of the CI compilers (*g++-10_serial_boost1.78.0_examples.TransitionRadiation*) does not recognize the arguments on the next line and assumes that there are 0 instead of 3 arguments)

```
/builds/hzdr/crp/picongpu/include/picongpu/plugins/transitionRadiation/TransitionRadiation.x.cpp:229:17: note: candidate: 'picongpu::plugins::transitionRadiation::TransitionRadiation<ParticlesType>::TransitionRadiation(std::shared_ptr<picongpu::plugins::multi::IHelp>&, size_t, picongpu::MappingDesc*) [with ParticlesType = picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting, picongpu::transitionRadiationMask> >; size_t = long unsigned int; picongpu::MappingDesc = pmacc::MappingDescription<3, pmacc::math::CT::Vector<std::integral_constant<int, 8>, std::integral_constant<int, 8>, std::integral_constant<int, 4> > >]'
  229 |                 TransitionRadiation(
      |                 ^~~~~~~~~~~~~~~~~~~
/builds/hzdr/crp/picongpu/include/picongpu/plugins/transitionRadiation/TransitionRadiation.x.cpp:229:17: note:   candidate expects 3 arguments, 0 provided
make[2]: *** [CMakeFiles/picongpu.dir/build.make:440: CMakeFiles/picongpu.dir/plugins/transitionRadiation/TransitionRadiation.x.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:207: CMakeFiles/picongpu.dir/all] Error 2
```

Retriggering the CI causes the same error.

Reply:
`PIC_REGISTER_SPECIES_PLUGIN(picongpu::plugins::multi::Master<picongpu::BinEnergyParticles<boost::mpl::_1>>);`

is an example how to use the macro at the end of this file. Each plugin must be constructable by `plugin()`

Reply:
That was the error!